src/metabase/feature_extraction/async.clj

```diff
@@ -46,17 +46,19 @@
 (defn compute
   "Compute closure `f` asynchronously. Returns id of the associated computation
    job."
-  [f]
-  (let [job (db/insert! ComputationJob
-                        :creator_id api/*current-user-id*
-                        :status     :running
-                        :type       :simple-job)
-        id  (:id job)]
-    (swap! running-jobs assoc id (future
-                                   (try
-                                     (save-result job (f))
-                                     (catch Exception e
-                                       (save-error job e)))))
+  [ctx f & args]
+  (let [id (hash (concat [ctx] args))]
```

Comment:
This is one of the main bits of complexity/bug surface area, this should definitely be pulled out into a function and carefully picked over. Ideally this hash/transform should be stable, and (bonus points) easy to parse by a human, so errors can be debugged or at least noticed.